packages/applications-service-api/__tests__/integration/applications.test.js

```diff
@@ -10,65 +12,95 @@ jest.mock('../../src/lib/prisma', () => ({
-				WebAddress: 'TBC',
-				sourceSystem: 'ODT',
-				stage5ExtensionToRecommendationDeadline: null
+				expect(response.status).toEqual(200);
+				expect(response.body).toEqual({
+					AnticipatedDateOfSubmission: '2023-09-01 00:00:00.0000000',
+					AnticipatedGridRefEasting: 485899,
+					AnticipatedGridRefNorthing: 414508,
+					AnticipatedSubmissionDateNonSpecific: 'Q3 2023',
+					ApplicantEmailAddress: 'TBC',
+					ApplicantPhoneNumber: 'TBC',
+					CaseReference: 'EN0110004',
+					ConfirmedDateOfDecision: null,
+					ConfirmedStartOfExamination: null,
+					DateOfDCOAcceptance_NonAcceptance: null,
+					DateOfDCOSubmission: null,
+					DateOfPreliminaryMeeting: null,
+					DateOfRecommendations: null,
+					DateOfRelevantRepresentationClose: null,
+					DateOfRepresentationPeriodOpen: null,
+					DateProjectWithdrawn: null,
+					DateRRepAppearOnWebsite: null,
+					DateTimeExaminationEnds: null,
+					LongLat: [-0.7028315466694124, 53.620079146110655],
+					MapZoomLevel: 6,
+					ProjectEmailAddress: 'drax.project.email@example.org',
+					ProjectLocation: 'Drax Power Station, North Yorkshire',
+					ProjectName: 'Drax Bioenergy with Carbon Capture and Storage Project',
+					PromoterFirstName: 'TBC',
+					PromoterLastName: 'TBC',
+					PromoterName: 'TBC',
+					Proposal: 'EN01 - Generating Stations',
+					Region: 'yorkshire_and_the_humber',
+					Stage: 1,
+					Stage4ExtensiontoExamCloseDate: null,
+					Stage5ExtensiontoDecisionDeadline: null,
+					Summary:
+						'Drax Power Limited proposes to install post-combustion capture technology that would capture carbon dioxide emissions from up to two of the existing biomass units at Drax Power Station. The proposal includes the construction and operation of carbon capture technology and associated equipment, and the integration of the units into the existing Common Services at Drax Power Station. The proposal includes associated development.',
+					WebAddress: 'TBC',
+					sourceSystem: 'ODT',
+					stage5ExtensionToRecommendationDeadline: null
+				});
+			});
+
+			it('returns 400 if caseReference is in invalid format', async () => {
+				const response = await request.get('/api/v1/applications/bad_format');
+
+				expect(response.status).toEqual(400);
+				expect(response.body).toEqual({
+					code: 400,
+					errors: ['\'caseReference\' must match pattern "^[A-Za-z]{2}\\d{6,8}$"']
+				});
 			});
 		});
+	});
+	describe('get all applications', () => {
+		it('happy path', async () => {
+			mockFindAll.mockResolvedValueOnce([{ dataValues: APPLICATION_DB }]);
```

Comment:
the project data in `APPLICATION_DB` is for a Back Office case, so it might be good to use some test data for a NI case instead. For example, `projectData` in the `__tests__/unit/controllers/applications.test.js` file

the fields used in the Back Office and NI databases are a bit different, so our API response will need to be consistent, so there will be some mapping to do - for example in the NI cases, the field `Stage` is an int, and from this API we want to return `stage` as an string